### PR TITLE
[Tracing Framework] Add proper API annotations

### PR DESCRIPTION
**Pull request by reta**
### Description
Add proper API annotations

### Related Issues
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@Gaganjuneja may I ask you to take a look please? thank you.

File: libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/TracerContextStorage.java [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
+
 /**
  * Storage interface used for storing tracing context
  * @param <K> key type
  * @param <V> value type
  *
  * @opensearch.internal
  */
+@InternalApi
 public interface TracerContextStorage<K, V> {
```
Comment (Gaganjuneja) [inline libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/TracerContextStorage.java:21]: Shouldn't it be @ExperimentalApi?

Comment (reta) [reply libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/TracerContextStorage.java:21]: We don't expose it anywhere, and only use internally in `DefaultTracer`

Comment (Gaganjuneja) [reply libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/TracerContextStorage.java:21]: ThreadContextBasedTracerContextStorage is internal one. There could be multiple implementations provided. Now we can keep it internal also.

**Review by Gaganjuneja — APPROVED** [review]

**Comment by reta** [comment]
@andrross @dblock could you help me folks to get this one in before 2.10 cut? thank you

**Review by dblock — APPROVED** [review]
I only took a cursory look, I imagine this has low chance of breaking anything ;)

**Pull request merged**